src/utils/transaction.js

```diff
@@ -0,0 +1,192 @@
+import _ from 'lodash';
+import { Address, AliasAction, LinkAction, TransactionType } from 'symbol-sdk';
+import { getFinanceBotPublicKeys } from '@src/config/environment';
+import { getPublicKeyFromPrivateKey } from '@src/utils/account';
+import { filterCurrencyMosaic } from '@src/utils/filter';
+import { getMosaicRelativeAmount } from '@src/utils/format';
+import { TransactionInfoPreviewValueType } from '@src/storage/models/TransactionInfoPreviewModel';
+import { Constants } from '@src/config/constants';
+
+/**
+ * Checks whether transaction is awaiting a signature by account.
+ */
+export const transactionAwaitingSignatureByAccount = (transaction, account, multisigAddresses) => {
+    if (transaction.transactionType === TransactionType.AGGREGATE_BONDED) {
+        const accountPublicKey = getPublicKeyFromPrivateKey(account.privateKey);
+        const transactionSignerAddresses = transaction.innerTransactions.map(innerTransaction => innerTransaction.signerAddress);
+        const cosignRequired = transactionSignerAddresses.some(
+            transactionSignerAddress => transactionSignerAddress && multisigAddresses?.some(address => address === transactionSignerAddress)
+        );
+        const transactionHasMissingSignatures = !!transaction.transactionInfo?.merkleComponentHash.startsWith('000000000000');
+        const signedByCurrentAccount = transaction.cosignaturePublicKeys.some(publicKey => publicKey === accountPublicKey);
+
+        return (
+            (!signedByCurrentAccount && transaction.status !== Constants.Message.CONFIRMED) ||
+            (transactionHasMissingSignatures && cosignRequired)
+        );
+    }
+
+    return false;
+};
```

Comment:
@bassemmagdy moved [this part](https://github.com/symbol/mobile-wallet/pull/230/files#diff-4ca36d50c878e4284c9d5c900368a8b61dc59e90bccf9433355f71b622da80ecL73-R101) here